release_dashboard/lib/services/release_dashboard_next_context.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/conductor_core.dart';
+import 'package:conductor_core/proto.dart' as pb;
+import 'package:file/file.dart' show File;
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../state/status_state.dart';
+
+/// Service class for using [NextContext] in the local release dashboard environment.
+class ReleaseDashboardNextContext extends NextContext {
+  ReleaseDashboardNextContext({
+    required this.context,
```

Comment:
Needs a widget context to access the global state with `context.read<StatusState>().syncStatusWithState();`

Reply:
I wonder if saving this will mess with the internals of Flutter? Rather than caching the `BuildContext`, you could just tear-off the `syncStatusWithState` function. Something like:

```
final void Function() syncStatusWithState;
```

And then when you create the context:

```
final ReleaseDashboardNextContext nextContext = ReleaseDashboardNextContext(
  syncStatusWithState: context.read<StatusState>().syncStatusWithState, // no parens here, not calling the function, but tearing it off and passing to the ReleaseDashboardNextContext constructor
  ... // the rest of the arguments
);
```

Reply:
Oh, that's a good point. It is specified [here](https://github.com/flutter/flutter/blob/18116933e77adc82f80866c928266a5b4f1ed645/packages/flutter/lib/src/widgets/framework.dart#L2020) that it would not be cached beyond the execution of a single synchronous function.